lld/wasm/Writer.cpp

```diff
@@ -473,6 +473,12 @@ void Writer::layoutMemory() {
     WasmSym::heapEnd->setVA(memoryPtr);
   }
 
+  if (config->maxMemory != 0 && config->noGrowableMemory) {
+    // Erroring out here is simpler than defining precedence rules.
+    error("--max-memory and --no-growable-memory are mutually exclusive");
+  }
```

Comment:
Can you do this error checking in `readConfigs`?

There are a bunch of other error message there of the form `xxx is incompatible with yyy`.  Can we also use that terminology for consistency?